client/src/components/tools/Tool.js

```diff
@@ -345,13 +346,14 @@ export default class Tool extends localization.LocalizedReactComponent {
         </Row>
       );
     }
+
     const doUpload = (file) => {
-      const MAX_FILE_SIZE_KB = 50;
+      const maximumFileSize = this.props.preferences.getPreferenceValue('misc.max.tool.icon.size.kb') || DEFAULT_FILE_SIZE_KB;
```

Comment:
The last thing: please define `getter` for this preference in `preferences` class (just like `useSpot`, `deploymentName` etc) with `undefined` value if it's not set **or not a number** and **converted to a number** value otherwise (we don't want to compare `string` and `number`).